packages/@sanity/core/src/util/checkRequiredDependencies.ts

```diff
@@ -0,0 +1,238 @@
+      This might cause problems!
+    `)
+  }
+}
+
+/**
+ * Read the declared peer dependencies of the installed `@sanity/base` module
+ *
+ * @param studioPath Path to the studio, in order to resolve `@sanity/base`
+ * @returns Object of peer dependencies (name => version), if any
+ */
+async function readBasePeerDependencies(
+  studioPath: string
+): Promise<Record<string, string | undefined>> {
+  const manifestPath = resolveFrom.silent(studioPath, path.join('@sanity/base', 'package.json'))
+
+  // If we can't resolve the manifest path, that means `@sanity/base` is not installed
+  if (!manifestPath) {
+    throw new Error(
+      'Failed to resolve @sanity/base/package.json - install dependencies with `npm install` or `yarn`.'
+    )
+  }
+
+  const {peerDependencies} = await readPackageManifest(manifestPath)
+  return peerDependencies
+}
+
+/**
+ * Reads the version number of the _installed_ module, or returns `null` if not found
+ *
+ * @param studioPath Path of the studio
+ * @param moduleName Name of module to get installed version for
+ * @returns Version number, of null
+ */
+async function readModuleVersion(studioPath: string, moduleName: string): Promise<string | null> {
+  const manifestPath = resolveFrom.silent(studioPath, path.join(moduleName, 'package.json'))
+  return manifestPath ? (await readPackageManifest(manifestPath)).version : null
+}
+
+/**
+ * Read the `package.json` file at the given path and return an object that guarantees
+ * the presence of name, version, dependencies, dev dependencies and peer dependencies
+ *
+ * @param packageJsonPath Path to package.json to read
+ * @returns Reduced package.json with guarantees for name, version and dependency fields
+ */
+async function readPackageManifest(packageJsonPath: string): Promise<PackageManifest> {
+  let manifest: unknown
+  try {
+    manifest = await readJSON(packageJsonPath)
+  } catch (err) {
+    throw new Error(`Failed to read "${packageJsonPath}": ${err.message}`)
+  }
+
+  if (!isPackageManifest(manifest)) {
+    throw new Error(`Failed to read "${packageJsonPath}": Invalid package manifest`)
+  }
+
+  const {name, version, dependencies = {}, devDependencies = {}, peerDependencies = {}} = manifest
+  return {name, version, dependencies, devDependencies, peerDependencies}
+}
+
+/**
+ * Install the passed dependencies at the given version/version range,
+ * prompting the user whether to use yarn or npm. If a `yarn.lock` file
+ * is found in the working directory, we will default the choice to yarn
+ *
+ * @param dependencies Object of dependencies (package name => version)
+ * @param context CLI context
+ */
+async function installDependenciesWithPrompt(
+  dependencies: Record<string, string>,
+  context: CliCommandContext
+): Promise<unknown> {
+  const {output, prompt, workDir, yarn} = context
+  const yarnLockExists = await fileExists(path.join(workDir, 'yarn.lock'))
```

Comment:
i typically reach for `fs.existsSync` for this kind of check. i don't know if there are any downsides to synchronous   file existence checks but node says it's not deprecated 🤷. no action requested for this one, just curious on your thoughts of if i shouldn't do that lol

Reply:
Nothing wrong with that except it blocks the event loop, so spinners and such will pause. Not a huge deal, but also don't see much of a downside in doing it async (except for the potential race condition of the file disappearing before the next operation, but that isn't really relevant in this case)